bulk_processing/new_address_processor.py

```diff
@@ -0,0 +1,94 @@
+from bulk_processing.bulk_processor import BulkProcessor
+from bulk_processing.processor_interface import Processor
+from bulk_processing.validators import mandatory, max_length, numeric, no_padding_whitespace, latitude_longitude, \
+    in_set, region_matches_treatment_code, ce_u_has_expected_capacity
+from config import Config
+
+
+class NewAddressProcessor(Processor):
+    TREATMENT_CODES = {
+        'HH_LP1E', 'HH_LP1W', 'HH_LP2E', 'HH_LP2W', 'HH_QP3E', 'HH_QP3W', 'HH_1ALSFN', 'HH_2BLEFN',
+        'HH_2CLEFN', 'HH_3DQSFN', 'HH_3EQSFN', 'HH_3FQSFN', 'HH_3GQSFN', 'HH_4HLPCVN', 'HH_SPGLNFN', 'HH_SPGQNFN',
+        'CE_LDIEE', 'CE_LDIEW', 'CE_LDIUE', 'CE_LDIUW', 'CE_QDIEE', 'CE_QDIEW', 'CE_LDCEE', 'CE_LDCEW',
+        'CE_1QNFN', 'CE_2LNFN', 'CE_3LSNFN', 'SPG_LPHUE', 'SPG_LPHUW', 'SPG_QDHUE', 'SPG_QDHUW', 'SPG_VDNEE',
+        'SPG_VDNEW'}
+
+    ESTAB_TYPES = {'HALL OF RESIDENCE', 'CARE HOME', 'HOSPITAL', 'HOSPICE', 'MENTAL HEALTH HOSPITAL',
+                   'MEDICAL CARE OTHER', 'BOARDING SCHOOL', 'LOW/MEDIUM SECURE MENTAL HEALTH',
+                   'HIGH SECURE MENTAL HEALTH', 'HOTEL', 'YOUTH HOSTEL', 'HOSTEL', 'MILITARY SLA', 'MILITARY US',
+                   'RELIGIOUS COMMUNITY', 'RESIDENTIAL CHILDRENS HOME', 'EDUCATION OTHER', 'PRISON',
+                   'IMMIGRATION REMOVAL CENTRE', 'APPROVED PREMISES', 'ROUGH SLEEPER', 'STAFF ACCOMMODATION',
+                   'CAMPHILL', 'HOLIDAY PARK', 'HOUSEHOLD', 'SHELTERED ACCOMMODATION', 'RESIDENTIAL CARAVAN',
+                   'RESIDENTIAL BOAT', 'GATED APARTMENTS', 'MOD HOUSEHOLDS', 'FOREIGN OFFICES', 'CASTLES', 'GRT SITE',
+                   'MILITARY SFA', 'EMBASSY', 'ROYAL HOUSEHOLD', 'CARAVAN SITE', 'MARINA', 'TRAVELLING PERSONS',
+                   'TRANSIENT PERSONS'}
+
+    file_prefix = Config.BULK_NEW_ADDRESS_FILE_PREFIX
+    routing_key = Config.NEW_ADDRESS_EVENT_ROUTING_KEY
+    exchange = ''
+    bucket_name = Config.BULK_NEW_ADDRESS_BUCKET_NAME
+    project_id = Config.BULK_NEW_ADDRESS_PROJECT_ID
+    schema = {
+
+        'UPRN': [mandatory(), max_length(12), numeric(), no_padding_whitespace()],
+        'ESTAB_UPRN': [mandatory(), max_length(12), numeric(), mandatory(), no_padding_whitespace()],
+        'ADDRESS_TYPE': [mandatory(), in_set({'HH', 'CE', 'SPG'}, label='address_types'), no_padding_whitespace()],
+        'ESTAB_TYPE': [mandatory(), in_set(ESTAB_TYPES, label='estab types'), max_length(255), no_padding_whitespace()],
+        'ADDRESS_LEVEL': [mandatory(), in_set({'E', 'U'}, label='address levels'), no_padding_whitespace()],
```

Comment:
we should probably boost the validation now that we're opening the floodgates to chaos monkeys - we can't have a 'U' level on an 'HH' address type, for example

Reply:
I'll create a card to look at the extra validation we could put onto this and in the sample validator. This is card was only meant for copying those over